pkg/kubectl/cmd/run.go

```diff
@@ -213,7 +214,9 @@ func Run(f cmdutil.Factory, cmdIn io.Reader, cmdOut, cmdErr io.Writer, cmd *cobr
 		}
 		switch restartPolicy {
 		case api.RestartPolicyAlways:
-			if contains(resourcesList, v1beta1.SchemeGroupVersion.WithResource("deployments")) {
+			if contains(resourcesList, appsv1beta1.SchemeGroupVersion.WithResource("deployments")) {
+				generatorName = "deployment/apps.v1beta1"
+			} else if contains(resourcesList, extensionsv1beta1.SchemeGroupVersion.WithResource("deployments")) {
 				generatorName = "deployment/v1beta1"
```

Comment:
I don't think changing the generator in this case is a breaking change. It's actually why we added the new defaults, so new deployments can take advantage of them. If the generator change was changing the resulting object kind that got created then that would be a breaking change, not this one. IMO